### PR TITLE
Refactor and extension

### PR DESCRIPTION
**Pull request by saturnboy**
huge refactor of the main `AssertionBuilder` class into separate mixin classes...this should help make things more readable and understandable to others.

### CODE REVIEW

**Comment by saturnboy** [comment]
@amitwer here is PR to completely reorganize the main assertion class into a bunch of sub-classes via the mixin pattern.  Feedback?

**Comment by saturnboy** [comment]
@alex-bes here is big refactor PR, including an extension system for welding in your own assertions.  See my test in 83e8bb51ff817c46fad1332bc2db8f1fc20d8390 for example usage. Feedback?

File: assertpy/assertpy.py [hunk 1]
```diff
@@ -156,948 +135,10 @@ def __init__(self, val, description='', kind=None, expected=None):
...
-                    type(e).__name__))
-
-        # didn't fail as expected, so raise
-        self._err('Expected <%s> to raise <%s> when called with (%s).' % (
-            self.val.__name__,
-            self.expected.__name__,
-            self._fmt_args_kwargs(*some_args, **some_kwargs)))
-
-### helpers ###
     def _err(self, msg):
```
Comment (dwyatte) [inline assertpy/assertpy.py]: `_err` is defined here, which uses `print`, so does assertpy.py need to do `from __future__ import print_function` opposed to the other files?

Comment (saturnboy) [reply assertpy/assertpy.py]: correct, but per #97 probably replace `print(out)` with `logger.warning(out)`

**Pull request merged**